ra-optimisation/rao-commons/src/main/java/com/farao_community/farao/rao_commons/linear_optimisation/iterating_linear_optimizer/IteratingLinearOptimizer.java

```diff
@@ -0,0 +1,170 @@
+        }
+    }
+
+    public String optimize(RaoData raoData) {
+        this.raoData = raoData;
+        bestVariantId = raoData.getInitialVariantId();
+        String optimizedVariantId;
+        for (int iteration = 1; iteration <= parameters.getMaxIterations(); iteration++) {
+            optimizedVariantId = raoData.cloneWorkingVariant();
+            raoData.setWorkingVariant(optimizedVariantId);
+            if (!optimize(iteration)
+                || !hasRemedialActionsChanged(optimizedVariantId, iteration)
+                || !evaluateNewCost(optimizedVariantId, iteration)
+                || !hasCostImproved(optimizedVariantId, iteration)) {
+                return getBestVariantWithSafeDelete(optimizedVariantId);
+            }
+            updateBestVariantId(optimizedVariantId);
+        }
+        return bestVariantId;
+    }
+
+    private boolean optimize(int iteration) {
+        // If optimization fails iteration can stop
+        try {
+            LOGGER.info(format(LINEAR_OPTIMIZATION_START, iteration));
+            simpleLinearOptimizer.optimize(raoData);
+            if (!simpleLinearOptimizer.getSolverResultStatusString().equals("OPTIMAL")) {
+                LOGGER.info(format(LINEAR_OPTIMIZATION_INFEASIBLE, iteration)); //handle INFEASIBLE solver status
+                return false;
+            }
+            LOGGER.info(format(LINEAR_OPTIMIZATION_END, iteration));
+            return true;
+        } catch (LinearOptimisationException e) {
+            LOGGER.error(format(LINEAR_OPTIMIZATION_ERROR, iteration, e.getMessage()));
+            return false;
+        }
+    }
+
+    private boolean hasRemedialActionsChanged(String optimizedVariantId, int iteration) {
+        // If the solution has not changed, no need to run a new sensitivity computation and iteration can stop
+        if (raoData.getRaoDataManager().sameRemedialActions(bestVariantId, optimizedVariantId)) {
+            LOGGER.info(format(SAME_RESULTS, iteration));
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    protected boolean evaluateNewCost(String optimizedVariantId, int iteration) {
+        // If evaluating the new cost fails iteration can stop
+        raoData.setWorkingVariant(optimizedVariantId);
+        try {
+            LOGGER.info(format(SYSTEMATIC_SENSITIVITY_COMPUTATION_START, iteration));
+            systematicSensitivityComputation.run(raoData);
+            raoData.getRaoDataManager().fillCracResultsWithSensis(simpleLinearOptimizer.getParameters().getObjectiveFunction(), systematicSensitivityComputation);
+            LOGGER.info(format(SYSTEMATIC_SENSITIVITY_COMPUTATION_END, iteration));
+            return true;
+        } catch (SensitivityComputationException e) {
+            LOGGER.error(format(SYSTEMATIC_SENSITIVITY_COMPUTATION_ERROR, iteration, systematicSensitivityComputation.isFallback() ? "Fallback" : "Default", e.getMessage()));
+            return false;
+        }
```

Comment:
I'm afraid if we do that to hide some errors that must not happen. For SensititivityComputationException we know that it might happen for network purpose and we want to continue after that but I feel that errors during filling results are not something that we can ignore

Reply:
Indeed, the error when filling the results is typically : you made the computation in DC and ask for a minMargin in A. I don't really get the difference between been caught here or not, but yeah it is an error that we want to return to the user. So maybe forget the initial comment